File: src/types/serverSession.ts
import type { LaunchConfig } from './recommendation';

export interface ServerSessionStatus {
  running: boolean;
  pid?: number | null;
  game_id?: string | null;
  profile_id?: string | null;
  ready: boolean;
  logs: string[];
  message: string;
  exit_code?: number | null;
  exited_at?: string | null;
  ever_ready: boolean;
  started_at?: string | null;
  uptime_seconds?: number | null;
}

export type ServerLaunchConfig = LaunchConfig;

export interface GenericServerLaunchConfig {
  game_name?: string | null;
  executable_path: string;
  working_dir?: string | null;
  port: number;
  args?: string[] | null;
  raw_args?: string | null;
  jar_memory_mb?: number | null;
}

export interface GenericServerPreflightCheck {
  id: string;
  level: 'ok' | 'warn' | 'block' | string;
  label: string;
  detail: string;
}

export interface GenericServerPreflightReport {
  ok: boolean;
  can_start: boolean;
  executable_path: string;
  working_dir?: string | null;
  executable_kind: string;
  port: number;
  summary: string;
  checks: GenericServerPreflightCheck[];
}

File: src-tauri/src/models/server_session.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ServerSessionStatus {
    pub running: bool,
    pub pid: Option<u32>,
    pub game_id: Option<String>,
    pub profile_id: Option<String>,
    pub ready: bool,
    pub logs: Vec<String>,
    pub message: String,
    pub exit_code: Option<i32>,
    pub exited_at: Option<String>,
    pub ever_ready: bool,
    pub started_at: Option<String>,
    pub uptime_seconds: Option<u64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GenericServerLaunchConfig {
    pub game_name: Option<String>,
    pub executable_path: String,
    pub working_dir: Option<String>,
    pub port: u16,
    pub args: Option<Vec<String>>,
    pub raw_args: Option<String>,
    pub jar_memory_mb: Option<u32>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GenericServerPreflightCheck {
    pub id: String,
    pub level: String,
    pub label: String,
    pub detail: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GenericServerPreflightReport {
    pub ok: bool,
    pub can_start: bool,
    pub executable_path: String,
    pub working_dir: Option<String>,
    pub executable_kind: String,
    pub port: u16,
    pub summary: String,
    pub checks: Vec<GenericServerPreflightCheck>,
}
